tests/test_model.py

```diff
@@ -0,0 +1,84 @@
+import pytest
+from shortlister import model
+from pathlib import Path
+
+path= Path("test_role")
+pickle_file_name = Path("shortlist.pickle")
+csv_file=Path("criteria.csv")
+
+def test_load_criteria():
+    criteria_result = model.load_criteria(path/csv_file)
+    expected = model.Criterion
+    for criterion in criteria_result:
+        assert type(criterion) is expected
+        assert criterion.name and criterion.description and criterion.scores is not None
+
+@pytest.mark.parametrize("folder_path,expected",
+                         [(path,["Emma Jones","Michael Davis","Sarah Thompson"]),
+                          ("non_existing_folder",[])])
+def test_load_applicants(folder_path,expected):
+    applicants = model.load_applicants(folder_path)
+    result = [applicant.name for applicant in applicants]
+    assert result == expected
+
+def test_load_role():
+    criteria = [model.Criterion(name="PhD",
+                              description="Degree or relevant experience",
+                              scores = ("Unsatisfactory","Moderate","Satisfactory","Excellent")),
+                model.Criterion(name="Research software",
+                              description="Authorship,development and maintenance",
+                              scores = ("Unsatisfactory","Moderate","Satisfactory","Excellent")),
+                model.Criterion(name="Best practices",
+                              description="Issue tracking, testing, documentation etc.",
+                              scores = ("Unsatisfactory","Moderate","Satisfactory","Excellent"))]
+    result = model.load_role(path,model.load_criteria(path/csv_file))
+    
+
+    expected = model.Role(job_title="test_role",
+                        job_id="0001",
+                        criteria=criteria)
+
+    assert result == expected
+
+def test_save_load():
+    c = [model.Criterion(name="PhD",
+                         description="Degree or relevant experience",
+                         scores = ("Unsatisfactory","Moderate","Satisfactory","Excellent")),
+         model.Criterion(name="Research software",
+                         description="Authorship,development and maintenance",
+                         scores = ("Unsatisfactory","Moderate","Satisfactory","Excellent")),
+         model.Criterion(name="Best practices",
+                         description="Issue tracking, testing, documentation etc.",
+                         scores = ("Unsatisfactory","Moderate","Satisfactory","Excellent"))]
+    
+    a = [model.Applicant(name="George Smith",
+                         cv="placeholder",
+                         scores={model.Criterion(name="PhD",
```

Comment:
The `scores` field in `Applicant` doesn't create new instances of `Criterion` for the keys of the dictionary. It uses the same Criterion instances that are stored in the Role. So, your Applicant should look like:

```
a = [model.Applicant(name="a1", cv="cv1", scores={c[0]: "Excellent", c[1]: "Satisfactory"}), ...]
```

i.e. the Criterion objects you created above are the ones that are used as the keys.